src/LogFileReader.cpp

```diff
@@ -75,11 +81,23 @@ bool LogFileReader::readAll(const QString& fileName)
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
     // TODO implement this first
+QStringList sections= line.split(SPLIT);
+if(sections.length()!=COLUMNS)
+{
+    return false;
+}
 
-    // This is here to the compiler happy. Otherwise the compile
-    // will have an error warning about an unused variable. Remove this
-    // when you use it.
-    Q_UNUSED(line);
-    Q_UNUSED(batteryData);
+QString timeString= sections.at(0);
+//Converts timeString to time for battery data
+batteryData.time= QTime::fromString(timeString,STRING_TIME_CONVERSION);
+//Takes voltage
+bool voltageOK;
+batteryData.voltage=sections.at(1).toDouble(&voltageOK);
+//Takes curent
+bool currentOK;
+batteryData.current=sections.at(2).toDouble(&currentOK);
+if(!voltageOK || !currentOK ||!batteryData.time.isValid())
```

Comment:
I would rewrite this section to be 
`if(!(voltageOK && currentOK && batteryData.time.isValid()))`